### PR TITLE
2 add linting and formatting rules for client 

### LINKED ISSUE #2: Add linting and formatting rules for client
* add eslint and necessary plugins for next tailwind and so on
* add prettier and its configuration
* dont forget to fix/add npm scripts

### PR DESCRIPTION
**Pull request by metehancelik**
closes #2

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,12 +3,14 @@
...
-    "dev": "npx nx dev server & npx nx dev client"
+    "dev": "run-p server client",
+    "server": "npx nx dev server",
+    "client": "npx nx dev client"
   },
   "private": true,
-  "dependencies": {},
   "devDependencies": {
     "@nx/js": "16.8.1",
+    "npm-run-all": "^4.1.5",
```
Comment (demirtasdurmus) [inline package.json]: @metehancelik  I think you don't need an external package to run apps in paralles. Nx also comes with a handy command to achieve this. Please remove this package and try the following command.
`npx nx run-many --target=dev --projects=server,client`

File: apps/client/package.json [hunk 2]
```diff
@@ -6,20 +6,36 @@
     "dev": "next dev",
     "build": "next build",
     "start": "next start",
-    "lint": "next lint"
+    "lint": "eslint \"src/**/*.{ts,tsx}\" --quiet",
```
Comment (demirtasdurmus) [inline apps/client/package.json:9]: When I run `npx nx lint client`, I get the following error. It seems you set the eslint rules but forgot to apply them into the boilerplate code :)
`➜ nx lint client                                   

> nx run client:lint

> client@0.1.0 lint
> eslint "src/**/*.{ts,tsx}" --quiet

/Users/durmusdemirtas/Documents/common-projects/weekend-blog/apps/client/src/app/page.tsx
   49:25  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
   55:24  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
   66:25  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
   72:24  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
   83:25  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
   89:24  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
  100:25  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence
  106:24  error  Curly braces are unnecessary here  react/jsx-curly-brace-presence

✖ 8 problems (8 errors, 0 warnings)
  8 errors and 0 warnings potentially fixable with the `--fix` option.

npm ERR! Lifecycle script `lint` failed with error: 
npm ERR! Error: command failed 
npm ERR!   in workspace: client@0.1.0 
npm ERR!   at location: /Users/durmusdemirtas/Documents/common-projects/weekend-blog/apps/client 

 ——————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————————

 >  NX   Ran target lint for project client (2s)
 
    ✖    1/1 failed
    ✔    0/1 succeeded [0 read from cache]`

**Review by demirtasdurmus — CHANGES_REQUESTED** [review]

File: package.json (hunk 1 above)
Comment (metehancelik) [reply package.json]: i tried this command but --target option was wrong. fixed it now.

File: apps/client/package.json (hunk 2 above)
Comment (metehancelik) [reply apps/client/package.json:9]: I left it on purpose, as I assumed that applying linting rules is another task.

File: apps/client/package.json [hunk 3]
```diff
@@ -6,20 +6,36 @@
...
+    "lint:fix": "eslint \"src/**/*.{ts,tsx}\" --fix",
+    "format": "prettier --write \"src/**/*.{ts,tsx}\""
   },
   "dependencies": {
     "@types/node": "20.6.2",
     "@types/react": "18.2.21",
     "@types/react-dom": "18.2.7",
     "autoprefixer": "10.4.15",
-    "eslint": "8.49.0",
     "eslint-config-next": "13.4.19",
```
Comment (demirtasdurmus) [inline apps/client/package.json]: I think I didnt notice this in my previous review. Please add all eslint and type related packages to dev dependencies

**Review by demirtasdurmus — APPROVED** [review]

**Pull request merged**